platform/android/src/thread.cpp

```diff
@@ -1,14 +1,21 @@
 #include <mbgl/util/logging.hpp>
 #include <mbgl/util/platform.hpp>
+#include <mbgl/platform/thread.hpp>
 
 #include <sys/prctl.h>
 #include <sys/resource.h>
 
+#include <cassert>
+#include "jni.hpp"
+
 // Implementation based on Chromium's platform_thread_android.cc.
 
 namespace mbgl {
 namespace platform {
 
+thread_local static JNIEnv* env;
+thread_local static bool detach;
```

Comment:
Is this needed? Looks to me it is useful for pairing `attach_jni_thread` with `detach_jni_thread`. We could not store it and have an assert on debug mode to check the value on each call.

Reply:
The idea here is that if a thread is already attached to the JVM by a different source, that source should be responsible for detaching it.